### DISCUSSION TITLE
Implement BigNumber support for random

### DISCUSSION THREAD

**Discussion by ellis** [josdejong/mathjs#2714] [Ideas]
In mathjs v3.6.0, `random()` crashes when passed a `BigNumber`.  The documentation seems to indicate that it should basically work:

> Important: BigNumber is not supported by the following functions: gcd, lcm, xgcd, arg, random, atan2. These functions will downgrade BigNumber to Number, and return a Number.

But when I enter the following (as a minimalist example) at the node prompt:

```
var math = require('mathjs')
math.random(math.bignumber(1))
```

it produces the following error message:

```
TypeError: Invalid argument in function random
    at Object.randFunctions.random (./node_modules/mathjs/lib/function/probability/distribution.js:78:19)
    at repl:1:6
    at REPLServer.defaultEval (repl.js:262:27)
    at bound (domain.js:287:14)
    at REPLServer.runBound [as eval] (domain.js:300:12)
    at REPLServer.<anonymous> (repl.js:431:12)
    at emitOne (events.js:82:20)
    at REPLServer.emit (events.js:169:7)
    at REPLServer.Interface._onLine (readline.js:212:10)
    at REPLServer.Interface._line (readline.js:551:8)
```

The exception gets thrown here:

```
          if ((min !== undefined && !isNumber(min)) || (max !== undefined && !isNumber(max))) {
            throw new TypeError('Invalid argument in function random');
          }
```

I guess `isNumber` returns `false` for BigNumbers...?  Or a coercion would need to take place before this check?

**Comment by josdejong** [comment]
Thanks for reporting, `random(bignumber(2))` should indeed return a number instead of throw an error.

I see we have to update this section in the documentation, since for example `gcd` and `lcm` already support BigNumbers.

**Comment by josdejong** [comment]
Sorry, I remember now that we downgrade nowhere BigNumbers to numbers anymore, that was the behavior in mathjs v2. That means that `random` should indeed return an error as long as it doesn't support BigNumbers. I've updated the docs accordingly.

Let's just focus on implementing BigNumber support for `random`, that will solve all related issues.

**Comment by josdejong** [comment]
Oops closed the issue accidentally
